downloadDirectory/src/main/java/org/reactome/release/downloadDirectory/MapOldStableIds.java

```diff
@@ -0,0 +1,125 @@
+package org.reactome.release.downloadDirectory;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.nio.file.StandardOpenOption;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class MapOldStableIds {
+	private static final Logger logger = LogManager.getLogger();
+	
+	private static Connection connect = null;
+	
+	public static void execute(String username, String password, String host, String releaseNumber) throws Exception {
+		
+		Statement statement = null;
+		ResultSet resultSet = null;
+		
+		logger.info("Running MapOldStableIds step");
+		// Need to use mysql driver to access stable_identifiers db
+		logger.info("Connecting to stable_identifers db...");
+		Class.forName("com.mysql.jdbc.Driver");
+		connect = DriverManager.getConnection("jdbc:mysql://" + host + "/stable_identifiers?" + "user=" + username + "&password=" + password);
+		statement = connect.createStatement();
+		resultSet = statement.executeQuery("SELECT identifier,instanceId FROM StableIdentifier");
+		
+		// Iterate through returned results of DB IDs and stable IDs 
+		logger.info("Mapping Old Stable IDs to Current Stable IDs...");
+		Map<String,ArrayList<String>> dbIdToStableIds = new HashMap<String,ArrayList<String>>();
+		List<String> dbIds = new ArrayList<String>();
+		while (resultSet.next()) 
+		{
+			String stableId = resultSet.getString(1);
+			String dbId = resultSet.getString(2);
+			
+			if (dbIdToStableIds.get(dbId) != null) 
+			{
+				dbIdToStableIds.get(dbId).add(stableId);
+			} else {
+				ArrayList<String> stableIds = new ArrayList<String>(Arrays.asList(stableId));
+				dbIdToStableIds.put(dbId, stableIds);
+				dbIds.add(dbId);
+			}
+		}
+		Collections.sort(dbIds);
+		
+		// Iterate through array of stable IDs associated with DB ID, splitting into human and non-human groups
+		List<ArrayList<Object>> hsaIds = new ArrayList<ArrayList<Object>>();
+		List<ArrayList<Object>> nonHsaIds = new ArrayList<ArrayList<Object>>();
+		for (String dbId : dbIds) 
+		{
+			List<String> stableIds = dbIdToStableIds.get(dbId);
+			Collections.sort(stableIds);
+			
+			// After sorting the first stable ID in the array is considered the primary ID.
+			// An Array of Arrays is used here, with each interior array's first element being 
+			// the primaryId string and the second element being an array of the remaining stable IDs.
```

Comment:
Could you give an example in the comments of how the array of array looks?  I'm just having trouble visualizing it.